kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java

```diff
@@ -87,7 +87,11 @@ public KafkaWorkItemHandler(String bootstrapServers,
                    keySerializerClass);
         config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
                    valueSerializerClass);
+        // it is needed to change the classloader to KIEURLClassLoader for dependencies to be resolved and then, set it back
+        ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+        Thread.currentThread().setContextClassLoader(KafkaProducer.class.getClassLoader());
         producer = new KafkaProducer<Long, String>(config);
+        Thread.currentThread().setContextClassLoader(oldClassLoader);      
```

Comment:
try - finally block The oldClassLoader cannot be hanging in there.

Reply:
right, done!